packages/ifpack2/src/Ifpack2_Relaxation_def.hpp

```diff
@@ -2165,45 +2175,37 @@ MTGaussSeidel (const Tpetra::MultiVector<scalar_type,local_ordinal_type,global_o
       X_colMap->doImport (*X_domainMap, *importer, Tpetra::CombineMode::INSERT);
     }
 
-    for (size_t indVec = 0; indVec < NumVectors; ++indVec) {
-      if (direction == Tpetra::Symmetric) {
-        KokkosSparse::Experimental::symmetric_gauss_seidel_apply
-        (mtKernelHandle_.getRawPtr(), A_->getNodeNumRows(), A_->getNodeNumCols(),
-            kcsr.graph.row_map, kcsr.graph.entries, kcsr.values,
-            Kokkos::subview(X_colMap->template getLocalView<MyExecSpace> (), Kokkos::ALL (), indVec),
-            Kokkos::subview(B_in->template getLocalView<MyExecSpace> (), Kokkos::ALL (), indVec),
-            zero_x_vector, update_y_vector, DampingFactor_);
-      }
-      else if (direction == Tpetra::Forward) {
-        KokkosSparse::Experimental::forward_sweep_gauss_seidel_apply
-        (mtKernelHandle_.getRawPtr(), A_->getNodeNumRows(), A_->getNodeNumCols(),
-            kcsr.graph.row_map,kcsr.graph.entries, kcsr.values,
-            Kokkos::subview(X_colMap->template getLocalView<MyExecSpace> (), Kokkos::ALL (), indVec ),
-            Kokkos::subview(B_in->template getLocalView<MyExecSpace> (), Kokkos::ALL (), indVec),
-            zero_x_vector, update_y_vector, DampingFactor_);
-      }
-      else if (direction == Tpetra::Backward) {
-        KokkosSparse::Experimental::backward_sweep_gauss_seidel_apply
-        (mtKernelHandle_.getRawPtr(), A_->getNodeNumRows(), A_->getNodeNumCols(),
-            kcsr.graph.row_map,kcsr.graph.entries, kcsr.values,
-            Kokkos::subview(X_colMap->template getLocalView<MyExecSpace> (), Kokkos::ALL (), indVec ),
-            Kokkos::subview(B_in->template getLocalView<MyExecSpace> (), Kokkos::ALL (), indVec),
-            zero_x_vector, update_y_vector, DampingFactor_);
-      }
-      else {
-        TEUCHOS_TEST_FOR_EXCEPTION(
-            true, std::invalid_argument,
-            prefix << "The 'direction' enum does not have any of its valid "
-            "values: Forward, Backward, or Symmetric.");
-      }
+    if (direction == Tpetra::Symmetric) {
+      KokkosSparse::Experimental::symmetric_gauss_seidel_apply
+      (mtKernelHandle_.getRawPtr(), A_->getNodeNumRows(), A_->getNodeNumCols(),
+          kcsr.graph.row_map, kcsr.graph.entries, kcsr.values,
+          X_colMap->template getLocalView<MyExecSpace> (),
```

Comment:
This should always be device, so you could just call the nontemplated method `getLocalViewDevice`.